### PR TITLE
Matt cardarelli persist visa types in redux

### PR DESCRIPTION
**Pull request by MattRC7**
Had to refactor the checkbox dropdown a bit to actually use the visibleTypes prop, also had to add some Redux actions and update the reducer for the visa types.

The actual Visa data is still a placeholder but it should be easy to replace with a fetch when we have real data.

### CODE REVIEW

File: src/redux/actions.js [hunk 1]
```diff
@@ -59,10 +60,17 @@ export function clearDistanceFilter() {
...
+    type: INITIALIZE_VISA,
+    visas
+  };
+}
+
 export function changeVisaFilter(visa) {
   return {
     type: CHANGE_VISA,
-    visa
+    visa: visa
```
Comment (mjsebast) [inline src/redux/actions.js]: Should keep the short hand

File: src/components/TopBar/provider-type-dropdown.js [hunk 2]
```diff
@@ -3,25 +3,22 @@ import CheckBoxDropdown from "../Dropdowns/checkbox-dropdown";
 
 const defaultSubheaderText = "Not Selected";
 export default class ProviderTypeDropdown extends React.Component {
-  state = { subheaderText: defaultSubheaderText };
```
Comment (p-somers) [inline src/components/TopBar/provider-type-dropdown.js]: This change is good but should probably be in its own PR though

File: src/redux/actions.js (hunk 1 above)
Comment (jonnyjohannes) [reply src/redux/actions.js]: WHAT THE F*CK, MATT!

**Review by p-somers — APPROVED** [review]
just a couple minor things, looks good other than those

File: src/components/TopBar/provider-type-dropdown.js (hunk 2 above)
Comment (MattRC7) [reply src/components/TopBar/provider-type-dropdown.js]: Okay, do you want me to revert just this file, and move these changes into a different PR?

File: src/redux/actions.js (hunk 1 above)
Comment (MattRC7) [reply src/redux/actions.js]: Not sure why I did that! I'll change it back

**Comment by MattRC7** [comment]
I made both of the requested changes. Please review

**Review by mjsebast — APPROVED** [review]
